src/k8s-engine/progressTracker.ts

```diff
@@ -1,7 +1,15 @@
 import * as K8s from './k8s';
 
 /**
- * ProgressTracker is used to track progress of multiple parallel actions.
+ * ProgressTracker is used to track the progress of multiple parallel actions.
+ * It invokes a callback that takes a progress object as input when one of those
+ * actions comes to a close. An "action" is effectively a Promise with some
```

Comment:
Technically, the numeric progress should be triggered before close, but I think this is fine.